test/wdio.conf.js

```diff
@@ -31,6 +31,89 @@ async function capabilityDefaults() {
   };
 }
 
+class GitHubStatus {
+  /**
+   * @param {Object} p
+   * @param {string} p.sha
+   * @param {string} p.repo
+   * @param {string} p.token
+   * @param {string} p.context
+   */
+  constructor({sha, repo, token, context}) {
+    this.sha = sha;
+    this.repo = repo;
+    this.token = token;
+    this.context = context;
+  }
+
+  setTarget(priority, target) {
+    if (this.targetPriority > priority) {
+      return;
+    }
+
+    this.targetPriority = priority;
```

Comment:
I find it so upsetting that an instance of a class can have state that isn't declared as part of the constructor, doesn't have an accessor—indeed, isn't annotated anywhere—and yet can be accessed from outside of the class itself.

```js
class Foo {
  constructor({bar}) {
    this.bar = bar;
  }

  baz() {
    this.baz = "bazzed"
  }
};

f = new Foo({bar: "barrr"});
f.bar; // "barrr"
f.baz();
f.baz; // "bazzed"
```

JavaScript really is a disaster, isn't it?

Reply:
I don't bite the hand that feeds.